### PR TITLE
[release-4.12] OCPBUGS-19837: Update staticpod file permissions to conform with CIS benchmarks

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR is a manual cherry-pick of these 3 PRs;
* https://github.com/openshift/library-go/pull/1570
* https://github.com/openshift/library-go/pull/1536
* https://github.com/openshift/library-go/pull/1546

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/openshift/api v0.0.0-20230213134911-7ba313770556
 	github.com/openshift/build-machinery-go v0.0.0-20220913142420-e25cf57ea46d
 	github.com/openshift/client-go v0.0.0-20220831193253-4950ae70c8ea
-	github.com/openshift/library-go v0.0.0-20230707145856-7efc9e03154a
+	github.com/openshift/library-go v0.0.0-20231002074440-3f69f773d102
```
Comment (p0lyn0mial) [inline go.mod:18]: the SHA looks good, points to 4-12 branch (https://github.com/openshift/library-go/commit/3f69f773d1025f30711ad99f8bea698d2d4cd9a6)

**Review by soltysh — APPROVED** [review]
/label backport-risk-assessed

**Comment by ardaguclu** [comment]
@gangwgr could you PTAL?. Thanks.

**Pull request merged**